overlord/configstate/config/helpers.go

```diff
@@ -97,8 +97,10 @@ func PatchConfig(snapName string, subkeys []string, pos int, config interface{},
 		// Raw replaces pristine on commit. Unpack, update, and repack.
 		var configm map[string]interface{}
 
-		if err := jsonutil.DecodeWithNumber(bytes.NewReader(*config), &configm); err != nil {
-			return nil, fmt.Errorf("snap %q option %q is not a map", snapName, strings.Join(subkeys[:pos], "."))
+		if config != nil {
```

Comment:
Did we have a bug here or is it because of calling PatchConfig from new contexts now? Till now PatchConfig was called from transaction.Set() which guaranteed config to be non-nil (and this is assumed in other switch-cases below), should we keep it that way and ensure non-nil by the callers?

Reply:
I rewrote this part, this is no longer needed